tools/uart_hw_test/uart_test_windows.c

```diff
@@ -0,0 +1,311 @@
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific
+* language governing permissions and limitations under the License.
+*
+******************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <windows.h>
+
+/****************************************************************************
+ * Preprocessor definitions
+ ****************************************************************************/
+#define UART_PATH_STR		"\\\\.\\COM%s"
+#define UART_ALT_PATH_STR	"COM%s"
+#define TTYPATH_LEN		25
+#define MAX_PORTNO_LENGTH 	10
+
+#define UART_TEST_COUNT		100
+
+#define UART_TEST_START_VAL	0xaa
+#define UART_TEST_VAL		0x55
+#define UART_TEST_STOP_VAL	0xbb
+
+#define UART_MAX_DELAY_MS	80
+#define UART_POLL_TIMEOUT_MS	2000
+
+#define HANDSHAKE_INTERVAL	100	/* milliseconds */
+#define HANDSHAKE_CNT_MAX	10
+/****************************************************************************
+ * Private Variables
+ ****************************************************************************/
+static int pass, fail;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+static void configure_handle(HANDLE handle, int baud)
+{
+	DCB dcbSerialParams = { 0 };
+    dcbSerialParams.DCBlength = sizeof(dcbSerialParams);
+
+    //Get the current serial port configuration
+    if (!GetCommState(handle, &dcbSerialParams)){
+        //Handle the error condition
+        printf("Failed to get serial port configuration\n");
+        return;
+    }
+
+    //set the desired configuration parameters
+    dcbSerialParams.BaudRate = baud; 
+    dcbSerialParams.ByteSize = 8; 
+    dcbSerialParams.Parity = NOPARITY;
+    dcbSerialParams.StopBits = ONESTOPBIT;
+
+    //apply the new configuration
+    if (!SetCommState(handle, &dcbSerialParams)){
+        //Handle the error condition
+        printf("Failed to set serial port configuration\n");
+        return;
+    }
+
+    //Set the timeouts
+    COMMTIMEOUTS timeouts = { 0 }; 
+    timeouts.ReadIntervalTimeout = MAXDWORD; 
+    timeouts.ReadTotalTimeoutConstant = 0;
+    timeouts.ReadTotalTimeoutMultiplier = 0;
+    timeouts.WriteTotalTimeoutConstant = 0;
+    timeouts.WriteTotalTimeoutMultiplier = 0;
+
+    //apply the new timeouts
+    if (!SetCommTimeouts(handle, &timeouts)){
+        //handle the error condition
+        printf("Failed to set serial port timeouts\n");
+        return;
+    }
+
+	FlushFileBuffers(handle);
+
+	DWORD mode;
```

Comment:
Same as above. Please include appropriate header for this.

Reply:
done